app/src/main/java/com/example/concordia_campus_guide/LocationFragment/LocationFragmentViewModel.java

```diff
@@ -64,23 +64,49 @@ private GeoJsonLayer initLayer(GoogleMap map, Context applicationContext){
         return layer;
     }
 
-    /**
-     * Generate the hall building overlays
-     * @return the generate ground overlay option
-     */
-    public void getBuildingOverlay( GeoJsonFeature feature){
+    public void setBuildingGroundOverlayOptions(Building building){
+        building.setGroundOverlayOption(new GroundOverlayOptions()
+                .position(building.getCenterCoordinates(), building.getWidth(), building.getHeight())
+                .image(BitmapDescriptorFactory.fromAsset("buildings_floorplans/"+building.getBuildingCode().toLowerCase()+"_"+building.getAvailableFloors()[building.getAvailableFloors().length-1].toLowerCase()+".png"))
+                .bearing(building.getBearing()));
+    }
+
+    public Building getBuildingFromGeoJsonFeature(GeoJsonFeature feature){
+        LatLng centerPos = getCenterPositionBuildingFromGeoJsonFeature(feature);
+
+        String[] floorsAvailable = getFloorsFromBuildingFromGeoJsonFeature(feature);
+
+        float building_width = -1;
+
+        if(feature.getProperty("width") != null)
+            building_width = Float.parseFloat(feature.getProperty("width"));
+
+        float building_height = -1;
+
+        if(feature.getProperty("height") != null)
+            building_height  = Float.parseFloat(feature.getProperty("height"));
+
+        float building_bearing = -1;
+        if (feature.getProperty("bearing") != null)
+            building_bearing = Float.parseFloat(feature.getProperty("bearing"));
+
+        String building_code = feature.getProperty("code");
+
+        return new Building(centerPos,floorsAvailable, building_code, building_width, building_height, building_bearing);
+    }
+
+    public LatLng getCenterPositionBuildingFromGeoJsonFeature(GeoJsonFeature feature){
```

Comment:
The mthd names could be shorter. We know from the paramerters that wtv info we're getting from is from a geoJsonFeature. I'd suggest a simple getBuildingCenter or computer parseBuildingCenter.